docs/rfcs/20220702-prometheus-extension.md

```diff
@@ -0,0 +1,292 @@
+        |                        |                             |                    |     
+        |                        |               sub SQL query result               |     
+        |                        | <------------------------------------------------|     
+        |                        |                             |                    |     
+        |                        |----.                        |                    |     
+        |                        |    | transform data format  |                    |     
+        |                        |<---'                        |                    |     
+        |                        |                             |                    |     
+        |                        | response remote read request|                    |     
+        |                        | ---------------------------->                    |     
+        |                        |                             |                    |     
+        |                        |                             |----.               |     
+        |                        |                             |    | compute       |     
+        |                        |                             |<---'               |     
+        |                        |                             |                    |     
+        |                        | response sub PromQL request |                    |     
+        |                        | <----------------------------                    |     
+        |                        |                             |                    |     
+        | response PromQL request|                             |                    |     
+        | <-----------------------                             |                    |     
+      Client              ,------+-------.                ,----+-----.           CeresDB  
+        ,-.               |Query_Frontend|                |Prometheus|           ,.-^^-._ 
+        `-'               `--------------'                `----------'          |-.____.-|
+        /|\                                                                     |        |
+         |                                                                      |        |
+        / \                                                                     |        |
+                                                                                '-.____.-'
+```
+
+## Comparison
+Both ways can achieve our initial requirements and are able to implement distributed execution in the future. 
+
+- Embedded `CeresDB`
+    - Pros.
+        - `CeresDB` feeds data to `Prometheus` directly, reducing some computation and transmission.
+    - Cons.
+        - Need to customize a `Prometheus` specific interface. 
+        - It's better to bind all three components together for simplicity.
+- Separated `CeresDB` cluster
+    - Pros.
```

Comment:
The pros of this way may be explained as a counterpart of the cons of the way above. e.g.:
- No need to customize `Promethues` remote storage interface in CeresDB because `Prometheus` extension is decoupled with `CeresDB`.
- The deployment only requires one `Query Frontend` along with `Prometheus` which is more lightweight and les invasive.